src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioActiveStatementProvider.cs

```diff
@@ -80,7 +74,8 @@ void CancelWork()
 
                                 if (activeStatementsResult.ErrorCode != 0)
                                 {
-                                    unexpectedError = activeStatementsResult.ErrorCode;
+                                    builders[runtimeIndex] = ArrayBuilder<ActiveStatementDebugInfo>.GetInstance(0);
+                                    completion.TrySetResult(builders.ToFlattenedImmutableArrayAndFree());
```

Comment:
>completion.TrySetResult(builders.ToFlattenedImmutableArrayAndFree()); [](start = 36, length = 69)

This needs to be conditional based on the number of pending runtimes:

`if (Interlocked.Decrement(ref pendingRuntimes) == 0)`

We don't want to complete the entire task if there are more runtimes to process. Only if the currently processed runtime is the last runtime we complete.